python/defs.bzl

```
@@ -41,6 +41,45 @@ def _add_tags(attrs):
         attrs["tags"] = [_MIGRATION_TAG]
     return attrs
 
+def _current_py_toolchain_impl(ctx):
+    toolchain = ctx.toolchains[ctx.attr._toolchain]
+
+    files = []
+    vars = {}
+
+    if toolchain.py3_runtime and toolchain.py3_runtime.interpreter:
+        files += [toolchain.py3_runtime.interpreter] + toolchain.py3_runtime.files.to_list()
+        vars = dict(vars.items() + {"PYTHON3": toolchain.py3_runtime.interpreter.path}.items())
+
+    if toolchain.py2_runtime and toolchain.py2_runtime.interpreter:
+        files += [toolchain.py2_runtime.interpreter] + toolchain.py2_runtime.files.to_list()
+        vars = dict(vars.items() + {"PYTHON2": toolchain.py2_runtime.interpreter.path}.items())
+
+    return [
+        toolchain,
+        platform_common.TemplateVariableInfo(vars),
+        DefaultInfo(
+            runfiles = ctx.runfiles(files = files),
+            files = depset(files),
+        ),
+    ]
```

Using the depset and runfiles appropriate constructors:

```suggestion
    direct = []
    transitive = []
    vars = {}

    if toolchain.py3_runtime and toolchain.py3_runtime.interpreter:
        direct.append(toolchain.py3_runtime.interpreter)
        transitive.append(toolchain.py3_runtime.files)
        vars["PYTHON3"] = toolchain.py3_runtime.interpreter.path

    if toolchain.py2_runtime and toolchain.py2_runtime.interpreter:
        direct.append(toolchain.py2_runtime.interpreter)
        transitive.append(toolchain.py2_runtime.files)
        vars["PYTHON2"] = toolchain.py2_runtime.interpreter.path

    files = depset(direct, transitive = transitive)
    return [
        toolchain,
        platform_common.TemplateVariableInfo(vars),
        DefaultInfo(
            runfiles = ctx.runfiles(transitive_files = files),
            files = files,
        ),
    ]
```